### PR TITLE
fix: version updates and changelog

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,260 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+After:
+
+```ts
+let providers = [
+    {
+        config: {
+            thirdPartyId: "google",
+            clients: [
+                { clientId: "clientid1", clientSecret: "..." },
+                { clientId: "clientid2", clientSecret: "..." },
```
Comment (rishabhpoddar) [inline CHANGELOG.md:110]: need clientType here?

File: lib/ts/version.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
-export const version = "14.1.3";
+export const version = "15.0.0";
 
 export const cdiSupported = ["3.0"];
```
Comment (rishabhpoddar) [inline lib/ts/version.ts:18]: change dashboard version to 0.7

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,260 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+        -   One of redirectURIInfo (for code flow) or oAuthTokens (for token flow) is required
+-   Updated `appleRedirectHandlerPOST`
+    -   to accept all the form fields instead of just the code
+    -   to use redirect URI encoded in the `state` parameter instead of using the websiteDomain config.
+    -   to use HTTP 303 instead of javascript based redirection.
+-   Updated `signInUp` recipe interface function in thirdparty with new parameters:
+    -   `oAuthTokens` - contains all the tokens (access_token, id_token, etc.) as returned by the provider
+    -   `rawUserInfoFromProvider` - contains all the user profile info as returned by the provider
+-   Added `manuallyCreateOrUpdateUser` recipe function in thirdparty recipe instead, to be used in place of `signInUp`
+
```
Comment (rishabhpoddar) [inline CHANGELOG.md:45]: list all functions that have changed in sigature - go recipe wise.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/version.ts (hunk 2 above)
Comment (sattvikc) [reply lib/ts/version.ts:18]: Done

File: CHANGELOG.md (hunk 3 above)
Comment (sattvikc) [reply CHANGELOG.md:45]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (sattvikc) [reply CHANGELOG.md:110]: Done

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,10 +7,402 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+## [15.0.0]
```
Comment (rishabhpoddar) [inline CHANGELOG.md:10]:
```suggestion
## [15.0.0] - 2023-07-19
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,10 +7,402 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+            -   to use redirect URI encoded in the `state` parameter instead of using the websiteDomain config.
+            -   to use HTTP 303 instead of javascript based redirection.
+    -   Session recipe changes
+        -   Added mandatory `tenantId` as first parameter to the following recipe exposed functions:
+            -   `createNewSession`
+            -   `createNewSessionWithoutRequestResponse`
+            -   `validateClaimsInJWTPayload`
+        -   Added mandatory `tenantId` in the input for the following recipe interface functions. If any of these functions are overridden, they need to be updated accordingly:
+            -   `createNewSession`
+            -   `getGlobalClaimValidators`
```
Comment (rishabhpoddar) [inline CHANGELOG.md:140]: third party recipe change section is not properly indented

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,10 +7,402 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+    -   Added optional param `tenantId` to `Session.getAllSessionHandlesForUser`. If tenantId is undefined, sessions handles across all tenants are returned
+-   Adds optional param `tenantId` to `getUserCount` which returns total count across all tenants if not passed.
+-   Adds protected prop `tId` to the accessToken payload
+-   Adds `includesAny` claim validator to `PrimitiveArrayClaim`
+
 ### Fixes
 
 -   Fixed an issue where certain Dashboard API routes would return a 404 for Hapi
 
+### Migration
```
Comment (rishabhpoddar) [inline CHANGELOG.md:172]: do not make it 1, 2, 3.. make it a `####`

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,10 +7,402 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+            ],
+        },
+    },
+];
+```
+
+4. Change in the implementation of custom providers
+
+-   All config is part of `ProviderInput`
+-   To update `authorisationRedirect.Params` dynamically, `getAuthorisationRedirectURL` must be overridden
```
Comment (rishabhpoddar) [inline CHANGELOG.md:246]: move it to the next point or move it from here entirely.

File: CHANGELOG.md [hunk 8]
```diff
@@ -7,10 +7,402 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+                userId: "id",
+                email: "email",
+                emailVerified: "email_verified",
+            },
+        },
+    },
+};
+```
+
+Note: the SDK will fetch the oauth2 endpints from the providered OIDC discovery endpoint url + '/.well-known/openid-configuration'
```
Comment (rishabhpoddar) [inline CHANGELOG.md:336]: typo

Comment (rishabhpoddar) [inline CHANGELOG.md:336]: Properly format.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (sattvikc) [reply CHANGELOG.md:10]: Done

File: CHANGELOG.md (hunk 8 above)
Comment (sattvikc) [reply CHANGELOG.md:336]: Done

Comment (sattvikc) [reply CHANGELOG.md:336]: Done

File: CHANGELOG.md (hunk 7 above)
Comment (sattvikc) [reply CHANGELOG.md:246]: Done

File: CHANGELOG.md (hunk 6 above)
Comment (sattvikc) [reply CHANGELOG.md:172]: Done

File: CHANGELOG.md (hunk 5 above)
Comment (sattvikc) [reply CHANGELOG.md:140]: Done

**Pull request merged**